xonsh/proc.py

```diff
@@ -0,0 +1,307 @@
+"""Interface for running Python functions as subprocess-mode commands.
+
+Code for several helper methods in the `ProcProxy` class have been reproduced
+without modification from `subprocess.py` in the Python 3.4.2 standard library.
+The contents of `subprocess.py` (and, thus, the reproduced methods) are
+Copyright (c) 2003-2005 by Peter Astrand <astrand@lysator.liu.se> and were
+licensed to the Python Software foundation under a Contributor Agreement.
+"""
+import io
+import os
+import sys
+import platform
+
+from threading import Thread
+from subprocess import Popen, PIPE, DEVNULL, STDOUT
+from collections import Sequence
+
+from xonsh.tools import redirect_stdout, redirect_stderr
+
+iswindows = platform.system() == 'Windows'
```

Comment:
PEP8: global variables should be named in all uppercase.  `IS_WINDOWS`.  Also a better place for this might be tools.py, because I can see it being used by many modules.